### DISCUSSION TITLE
Visualizing iterative algorithms as they run

### DISCUSSION THREAD

**Discussion by FeldrinH** [fonsp/Pluto.jl#1303] [Pluto development]
Pluto is in general great and very convenient, however, one thing I struggle with is how to deal with visualizing iterative algorithms.

Say for example I have some iterative optimization algorithms, that optimize the current result with every iteration and can run indefinitely.

Since these algorithms can run indefinitely, I would like to somehow visualize the current state of the algorithm as it runs so I can monitor the progress while I wait as well as decide when to stop optimizing.
However, given that Pluto only shows the results of cells and there is no global state allowed, I am not sure how I should visualize intermediate results while the algorithms is running.

Are there any existing or planned methods/approaches for dealing with this sort of thing?

**Comment by Ellipse0934** [comment]
You can wrap your iteration in a `Task` and send intermediate data outside using a channel. You can use another channel to signal the Task to stop.

**Reply by FeldrinH** [reply]
That definitiely could work, however one concern I have is what if I start the task and then accidentally through some reactive update or code modification rerun the cell where the task is created. Then I will have lost the reference to the original task, however, the task will still be running and consuming system resources.

**Reply by Ellipse0934** [reply]
Declare an array of tasks in another cell and whenever you create a task push it to this array. If you manage to pull of something nicely with tasks I'm sure people would find it useful here as reference/educational material.
